test/integration/targets/template/tasks/main.yml

```diff
@@ -619,5 +619,28 @@
       - 'template_results.mode == "0547"'
       - 'stat_results.stat["mode"] == "0547"'
 
+# Test output_encoding
+- name: Prepare the list of encodings we want to check, including empty string for defaults
+  set_fact:
+    template_encoding_1252_encodings: ['', 'utf-8', 'windows-1252']
+
+- name: Copy known good encoding_1252_*.expected into place
+  copy:
+    src: 'encoding_1252_{{ item | default("utf-8", true) }}.expected'
+    dest: '{{ output_dir }}/encoding_1252_{{ item }}.expected'
+  loop: '{{ template_encoding_1252_encodings }}'
+
+- name: Generate the encoding_1252_* files from templates using various encoding combinations
+  template:
+    src: 'encoding_1252.j2'
+    dest: '{{ output_dir }}/encoding_1252_{{ item }}.txt'
+    output_encoding: '{{ item }}'
```

Comment:
Ah, I think you had the extra ```or``` in the code to handle this test setting output_encoding="".  Instead of changing the code like that, use this construct in the test's playbook:
```
- name: Generate the encoding_1252_* files from templates using various encoding combinations
  template:
    src: 'encoding_1252.j2'
    dest: '{{ output_dir }}/encoding_1252_{{ item }}.txt'
    output_encoding: '{{ item | default(omit, true) }}'
```

Reply:
Actually, the extra or in template.py is to handle all potential cases: even if the output_encoding is sent as None or empty string from a template, I would like the internal to be an explicit 'utf-8'. This way, we keep to the pythonic way of 'explicit is better than implicit', and avoid side-effects with the rest of the code (we don't rely on other parts of the code using utf-8 as the default encoding).

Of course, this is my reasoning, but you have the final word.
Let me know if you still want me to remove the 'or' construction.